dependencies.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# $1 MUST be 'init' or 'update'.
+
+
+# get submodules
+SM_CONFIG=$(cat submodules/ocrd_all/.gitmodules)
+
+SUBMODULE_NAMES=$(
+    for LINE in $SM_CONFIG
+    do
+        echo $LINE | grep -oE '"(.+?)"' | cut -d'"' -f2
+    done
+)
+
+CURRENT_DIR=$PWD
+
+# get the main OCR-D dependencies which stem from core and that most processors have in common.
+cd 'submodules/ocrd_all/core' || exit
+python -m venv venv
+VENV=$PWD'/venv/bin/python'
+CMD='-m pip install ocrd'
+$VENV $CMD
+CMD='-m pip freeze -l'
+$VENV $CMD > $CURRENT_DIR'/core_deps.txt'
+cd $CURRENT_DIR || exit
+
+echo '{' >> $CURRENT_DIR'/deps.json'
+
+for NAME in $SUBMODULE_NAMES
+do
+    if [ $NAME != 'tesseract' ] && [ $NAME != 'opencv-python' ]; then
+        DIR='/submodules/ocrd_all/'$NAME
+        echo "Currently processing " $NAME " ..."
+        cd $CURRENT_DIR$DIR || exit
+        python -m venv venv
+        VENV=$PWD'/venv/bin/python'
+
+        # install pkgs
+        if [ -f 'requirements.txt' ]; then
+            CMD_INSTALL=$(echo '-m pip install -r requirements.txt .')
+            $VENV $CMD_INSTALL
+        elif [ -f 'setup.py' ]; then
+            CMD_INSTALL=$(echo '-m pip install .')
+            $VENV $CMD_INSTALL
+        fi
+
+        # get deps and save them to JSON
+        CMD_FREZE=$(echo '-m pip freeze -l')
+        echo '"'$NAME'": {' >> $CURRENT_DIR'/deps.json'
+        RESULT=$($VENV $CMD_FREZE)
+        for RES in $RESULT
+        do
+            # check if dependency also occurs in core_deps.txt
+            IS_CORE_DEP="false"
+            for LINE in $(cat $CURRENT_DIR/core_deps.txt)
+            do
+                if [ $RES == $LINE ]; then
+                    IS_CORE_DEP="true"
+                fi
+            done
+
+            if [ $IS_CORE_DEP == "false" ]; then
+                PKG=$(echo $RES | cut -d'=' -f1)
+                VER=$(echo $RES | cut -d'=' -f3)
+
+                # in each venv the project itself is also installed.
+                # since it's not a dependency, we omit it.
+                PKG_NORMALIZED=$(echo "$PKG" | sed -e 's/-/_/g')
+                STARTS_WITH_FILE=$(echo "$PKG" | grep -E '^file')
```

Comment:
What's the purpose of checking whether pkg starts with `file`?

Reply:
At some point I had packages in my `deps.json` that looked like `ocrd-asdf@file//path/on/my/filesystem/to/venv`. But a thorough rebuild showed it was clutter that built up during development. Thanks for pointing that out and making me retest! I'll remove it.

Reply:
See https://github.com/OCR-D/kwalitee-dashboard-back-end/pull/14/commits/b4803bc54358dcd4621f6a7fe31ed48a08c4b886